apps/sr-frontend/src/app/Character.tsx

```diff
@@ -1,21 +1,26 @@
-import { CardThemed } from '@genshin-optimizer/common/ui'
-import type { AscensionKey } from '@genshin-optimizer/sr/consts'
+import { CardThemed, DropdownButton } from '@genshin-optimizer/common/ui'
+import { allEidolonKeys, type AscensionKey } from '@genshin-optimizer/sr/consts'
 import { convert, selfTag } from '@genshin-optimizer/sr/formula'
 import {
+  AbilityDropdown,
   BuildDisplay,
   useCalcContext,
   useCharacter,
   useCharacterContext,
   useCharacterReducer,
 } from '@genshin-optimizer/sr/ui'
-import { ExpandMore } from '@mui/icons-material'
+import { CheckBox, CheckBoxOutlineBlank, ExpandMore } from '@mui/icons-material'
 import {
   Accordion,
   AccordionDetails,
   AccordionSummary,
   Box,
+  Button,
   CardContent,
   Container,
+  Divider,
+  Grid,
+  MenuItem,
   Stack,
   TextField,
   Typography,
```

Comment:
For the `Character` component, it might make sense not to show this UI at all until there is a valid `characterKey`. This should prevent all the `character?.` checks.

Reply:
In terms of UX flow of character editor, you'd probably want to just show all the options, but with them disabled, please disregard previous comment.